lib/WebpackOptionsDefaulter.js

```diff
@@ -213,7 +213,7 @@ class WebpackOptionsDefaulter extends OptionsDefaulter {
 				}).apply(compiler);
 			}
 		}]);
-		this.set("optimization.nodeEnv", "make", options => options.mode);
+		this.set("optimization.nodeEnv", "make", options => options.mode || "production");
```

Comment:
I'm not a fan of `NODE_ENV` being set to `mode`. I know we don't advertise on `mode=none` but it's a thing. What about changing how it works to something more elaborated?

Reply:
This is part of the production mode behavior. We could hash it out in separate issue/PR. But this just repairs the intended capability to use `mode: production` behavior by default.

Reply:
Then it's fine 👍